extra/Addons/Macros/Panel.SpaceToSelect.lua

```diff
@@ -1,6 +1,10 @@
+local APANEL,INVERT,MODE_IDX,CUR_ITEM = 0,2,1,0
 Macro {
   description="Panel: Use Space to select files";
-  area="Shell"; key="Space";
+  area="Shell";
+  key="Space";
   condition=function() return CmdLine.Empty and APanel.Visible end;
-  action=function() Keys('Ins') end;
+  action=function()
+    Panel.Select(APANEL,INVERT,MODE_IDX,CUR_ITEM)
```

Comment:
Reverted.
Please don't smuggle breaking changes.

Reply:
It was not breaking change, because it is not part of code. It is just a sample macro. Sample config.
But whatever. I do not care anymore.

Reply:
"Breaking" as in "breaking users' habits".

Remember that people are lazy and don't want to learn anything.
For you it's an example, a building block/inspiration for something larger, for laypeople it's a black box of magic.

I'm not saying it's set in stone and we can't change any user-observable behavior, but it's usually better to discuss and at least do it separately from formatting / refactoring.

Reply:
> "Breaking" as in "breaking users' habits".

I still insist that changes in Addons do not break habits.
If someone already has macro for Space - it does not break.


> Remember that people are lazy and don't want to learn anything.

Far is not for so lazy people, as I can observe.
Because every month everything is changing in Far that brokes someones habits. And that was never seen as such a big issue.

> [...] but it's usually better to discuss and at least do it separately from formatting / refactoring.

I can agree that discussing it first is always better. That's why I created the forum's topics (though there is usually no noticeable community feedback).

BTW my PR introduced changes in behavior of some other macros too. So one can say that there were mixed formatting, refactoring, and functional changes.
But every commit was separate, so I do not see real issue here.

Reply:
> If someone already has macro for Space - it does not break.

Indeed, but from time to time people set it up from scratch.
Also, if I need to work on a new machine and don't have access to my profile, I might just run the installer and copy a couple of macros like Esc=CtrlO, Del=F8 and Space=Ins to make it borderline comfortable because I have neither time nor inspiration for more.

> every month something is changing in Far that brokes someones habits. And that was never seen as such a big issue.

Probably not every month, but yes, sometimes it happens, sometimes unintentionally, sometimes inevitably. But if we can avoid breaking things, then why break them?

> I created the forum's topics (though there is usually no noticeable community feedback)

I believe in this particular case there was a feedback. And, since you didn't object back then, we supposedly reached a concensus.

> my PR introduced changes in behavior of some other macros too

Oh. Not too dramatic changes hopefully?

> one can say that there were mixed formatting, refactoring, and functional changes.

Maintainer's POV:

I see a new PR.
I see it's something about formatting & refactoring in macros.
I see it's from John Doe.
I know that John Doe is a valuable and respectable member of the community and a long-time contributor who knows way more about macros than I do or ever will.
Therefore, I do not need to spend hours reviewing every single line and read 40 commit messages - I can merge it straight away and attend to my evening affairs.

But changes like "Adjusted things to my personal preferences" do need more attention and thus belong to a separate PR.
[Obligatory XKCD](https://xkcd.com/1172/).

> I still insist that changes in Addons do not break habits.

Macros like Panel.Esc, Panel.Del, Panel.BS and Panel.SpaceToSelect are not exactly some random examples, they are primordial interface options moved to macros a quarter of a century ago:
![image](https://github.com/FarGroup/FarManager/assets/11453922/b79cacbe-8e51-4119-b701-e88bf255232e)

They always worked the way they do.
People expect them to work the way they do.
They are probably mentioned in more than one FAQ.
Someone probably deploys them automatically to new workstations.
And someone probably asked about them in CS exam questions.
That's why I suggested a new macro back in 2015 and still think it's the right thing to do.

On top of that, the Ins key is heavily neglected on modern keyboards, so it's only natural for this macro to replicate its behavior exactly, rather than *almost, but not quite, entirely unlike*.